docs/source/architecture/data_services_architecture.rst

```diff
@@ -1,4 +1,109 @@
-Data Services Architecture
-==========================
+Motech Data Services Architecture
+=================================
 
-There will be text here.
+MOTECH Data Services (MDS) is responsible for creation of Data Models for MOTECH Platform either using Developer Defined Entities (DDE) or End User Defined Entities (EUDE). DDEs use custom annotations of MDS. The EUDEs can be created as well as configured from UI along with their Data Access Layer. Centre of Data Access Layer is Generic Data Access Pattern Class which defines generic persistence related methods.
```

Comment:
"DDEs use custom annotations of MDS" is not very clear. Maybe something like "DDEs use custom annotations of Java objects to specify entities and fields"...